packages/@glimmer/reference/test/iterable-test.ts

```diff
@@ -1,46 +1,38 @@
 import { module, test } from './utils/qunit';
 
-import { IterableReference, OpaqueIterationItem, UNDEFINED_REFERENCE } from '..';
+import {
+  createIteratorRef,
+  createComputeRef,
+  OpaqueIterationItem,
+  Reference,
+  valueForRef,
+} from '..';
 import { symbol } from '@glimmer/util';
 import { testOverrideGlobalContext, GlobalContext } from '@glimmer/global-context';
 import { VOLATILE_TAG, consumeTag } from '@glimmer/validator';
 
 import { TestContext } from './utils/template';
 import objectValues from './utils/platform';
 
-class VolatileReference<T> {
-  constructor(public inner: T) {}
-
-  isConst() {
-    return false;
-  }
-
-  value() {
-    consumeTag(VOLATILE_TAG);
-    return this.inner;
-  }
-
-  get(_key: string) {
-    return UNDEFINED_REFERENCE;
-  }
-}
-
 class IterableWrapper {
-  private iterable: IterableReference;
+  private iterable: Reference<{ next(): OpaqueIterationItem | null }>;
 
   constructor(obj: unknown, key = '@identity') {
-    let valueRef = new VolatileReference(obj);
-    this.iterable = new IterableReference(valueRef, key);
+    let valueRef = createComputeRef(() => {
+      consumeTag(VOLATILE_TAG);
```

Comment:
Why? This seems like an odd thing to test, since AFAICT none of our real environments actually use VOLTATILE_TAG like this.

Reply:
yeah, I think this should probably be updated at some point, but for now it's just a quick way to avoid caching